hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/BlockNodeConnectionManager.java

```diff
@@ -522,6 +535,55 @@ public boolean selectNewBlockNodeForStreaming(final boolean force) {
         return selectedNode;
     }
 
+    /**
+     * Task that creates a service connection to a block node and retrieves the status of the block node.
+     */
+    class RetrieveBlockNodeStatusTask implements Callable<BlockNodeStatus> {
+
+        private final BlockNodeServiceConnection svcConnection;
+
+        RetrieveBlockNodeStatusTask(@NonNull final BlockNodeConfiguration nodeConfig) {
+            requireNonNull(nodeConfig, "Node configuration is required");
+            svcConnection =
+                    new BlockNodeServiceConnection(configProvider, nodeConfig, blockingIoExecutor, clientFactory);
+        }
+
+        @Override
+        public BlockNodeStatus call() throws Exception {
+            svcConnection.initialize();
+
+            try {
+                return svcConnection.getBlockNodeStatus();
+            } finally {
+                svcConnection.close();
+            }
+        }
+    }
+
+    /**
+     * Converts a {@link Future} to a {@link CompletableFuture}.
+     * @param future the Future to convert
+     * @return a CompletableFuture that wraps the original Future
+     */
+    private <T> CompletableFuture<T> toCompletableFuture(final Future<T> future) {
```

Comment:
If the task `Future` objects are going to be converted to `CompletableFuture`s later anyway, why not just store `CompletableFuture` objects in `tasksByNode`? I don't see what advantage there is to using `Future` and then doing this awkward conversion.